[cli] Launch fails for "bash pmd"

Tthis is the problem:

```shell
PMD_HOME=/home/andreas/PMD/binaries/pmd-bin-7.0.0-rc3
PATH=${PMD_HOME}/bin:$PATH
bash pmd

/home/andreas/PMD/binaries/pmd-bin-7.0.0-rc3/bin/pmd: line 62: pushd: ./../lib: No such file or directory
/home/andreas/PMD/binaries/pmd-bin-7.0.0-rc3/bin/pmd: line 64: popd: directory stack empty
/home/andreas/PMD/binaries/pmd-bin-7.0.0-rc3/bin/pmd: line 84: pushd: ./../conf: No such file or directory
/home/andreas/PMD/binaries/pmd-bin-7.0.0-rc3/bin/pmd: line 86: popd: directory stack empty
Error: Could not find or load main class net.sourceforge.pmd.cli.PmdCli
Caused by: java.lang.ClassNotFoundException: net.sourceforge.pmd.cli.PmdCli
```

A PR exists for this in #4640

--- a/pmd-dist/src/main/resources/scripts/pmd
+++ b/pmd-dist/src/main/resources/scripts/pmd
@@ -52,7 +52,7 @@ set_lib_dir() {
     if [ -L "$0" ]; then
       local script_real_loc=$(readlink "$0")
     else
-      local script_real_loc=$0
+      local script_real_loc=${BASH_SOURCE[0]}
     fi
     local script_dir=$(dirname "${script_real_loc}")
 

--- a/pmd-dist/src/main/resources/scripts/pmd
+++ b/pmd-dist/src/main/resources/scripts/pmd
@@ -52,7 +52,7 @@ set_lib_dir() {
     if [ -L "$0" ]; then
       local script_real_loc=$(readlink "$0")
     else
-      local script_real_loc=${BASH_SOURCE[0]}
+      local script_real_loc=${BASH_SOURCE[0]:-${(%):-%x}}
     fi
     local script_dir=$(dirname "${script_real_loc}")
 

